templates/vdbench_save.yml

```diff
@@ -0,0 +1,160 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# https://aka.ms/yaml
+
+steps:
+- script: |
+    export PATH=$PATH:${BUILD_SOURCESDIRECTORY}/test/utils
+
+    az login --service-principal -u $ARM_CLIENT_ID -p $ARM_CLIENT_SECRET --tenant $ARM_TENANT_ID &&
+    az account set -s $ARM_SUBSCRIPTION_ID
+    check_rc.sh $? "Login to Azure and set default subscription" || exit 0
+
+    # Wait for the vdbench.py script to complete.
+    cmdokmsg="CMD_OK_$(date -u '+%m%dx%H%M%S')"
+    cmdlog=/tmp/vm_runcmd.log
+    polling_secs=60
+    while [ 1 ]; do
+      az vmss run-command invoke                      \
+        --resource-group ${RESOURCE_GROUP_NAME}-vmss  \
+        --name           "vmss"                       \
+        --instance-id    0                            \
+        --command-id     RunShellScript               \
+        --scripts        "runuser -l azureuser -c 'tmux ls' || echo ${cmdokmsg}" \
+        --query 'value[0].message' --output tsv |& tee ${cmdlog}
+      check_rc.sh ${PIPESTATUS[0]} "Checking for any tmux session " || exit
+      grep -q ${cmdokmsg} ${cmdlog}
+      if [ $? -ne 0 ]; then
+        echo "INFO: tmux session still active. Re-check in ${polling_secs} seconds."
+        sleep ${polling_secs}
+        continue
+      else
+        echo "INFO: tmux session no longer active"
+        break
+      fi
+    done
+  displayName: "Check for vdbench.py completion"
+  timeoutInMinutes: 360
```

Comment:
This is a six-hour timeout but looking at the latest successful run, this takes 15 minutes. Is this expected to take longer? The last successful run didn't do scale up/down, so maybe it should be this fast. Unsure, but it'd be good to clarify those expectations. I'd suggest adding a general PR comment linking to runs under the applicable configurations (e.g., with and without scale up/down, etc.).

Reply:
Will lower.

Reply:
Is ~15 minutes confirmed as the expected duration for the vdbench test in all cases?

Particularly curious about the scale up/down workflow since that process takes > 15 minutes (at least for a large scale down).

Reply:
That timeout is just for the vdbench check. I moved it down to 45 minutes. When running it without scaleup/scaledown past few days the max I saw was around 34 minutes. If scaleup/scaledown are ran, it is very quick because vdbench is usually already done. If scaleup/scale down are ran, it waits for those to complete before getting to the vdbench_save.yml template to look for vdbench results.

Reply:
Understood, but I wanted to know if the vdbench test duration is expected. So if scale up/down is run but vdbench completes before the scale up/down is complete, is that expected? (I thought we wanted to measure perf impact during scale up/down.)

This is likely something that can be addressed in a PBI this sprint. Resolving this comment.